src/test/storage/encoded_string_segment_test.cpp

```diff
@@ -0,0 +1,301 @@
+  }
+
+  std::shared_ptr<ValueSegment<pmr_string>> create_string_value_segment() {
+    auto values = pmr_concurrent_vector<pmr_string>(row_count);
+
+    for (auto i = 0u; i < row_count; ++i) {
+      if (i % 3 == 0) {
+        values[i] = "Hello world!!1!12345";
+      } else if (i % 3 == 1) {
+        values[i] = "This IS A ...";
+      } else {
+        values[i] = "0987654312poiuytrewq";
+      }
+    }
+
+    return std::make_shared<ValueSegment<pmr_string>>(std::move(values));
+  }
+
+  std::shared_ptr<ValueSegment<pmr_string>> create_string_w_null_value_segment() {
+    auto values = pmr_concurrent_vector<pmr_string>(row_count);
+    auto null_values = pmr_concurrent_vector<bool>(row_count);
+
+    std::default_random_engine engine{};
+    std::bernoulli_distribution bernoulli_dist{0.3};
+
+    for (auto i = 0u; i < row_count; ++i) {
+      null_values[i] = bernoulli_dist(engine);
+      if (i % 3 == 0) {
+        values[i] = "Hello world!!1!12345";
+      } else if (i % 3 == 1) {
+        values[i] = "This IS A ...";
+      } else {
+        values[i] = "0987654312poiuytrewq";
+      }
+    }
+
+    return std::make_shared<ValueSegment<pmr_string>>(std::move(values), std::move(null_values));
+  }
+
+  std::shared_ptr<PosList> create_sequential_position_filter() {
+    auto list = std::make_shared<PosList>();
+    list->guarantee_single_chunk();
+
+    std::default_random_engine engine{};
+    std::bernoulli_distribution bernoulli_dist{0.5};
+
+    for (auto offset_in_referenced_chunk = 0u; offset_in_referenced_chunk < row_count; ++offset_in_referenced_chunk) {
+      if (bernoulli_dist(engine)) {
+        list->push_back(RowID{ChunkID{0}, offset_in_referenced_chunk});
+      }
+    }
+
+    return list;
+  }
+
+  std::shared_ptr<PosList> create_random_access_position_filter() {
+    auto list = create_sequential_position_filter();
+
+    auto random_device = std::random_device{};
+    std::default_random_engine engine{random_device()};
+    std::shuffle(list->begin(), list->end(), engine);
+
+    return list;
+  }
+
+  template <typename T>
+  std::shared_ptr<BaseEncodedSegment> encode_value_segment(DataType data_type,
+                                                           const std::shared_ptr<ValueSegment<T>>& value_segment) {
+    const auto segment_encoding_spec = GetParam();
+    return encode_segment(segment_encoding_spec.encoding_type, data_type, value_segment,
+                          segment_encoding_spec.vector_compression_type);
+  }
+};
+
+auto formatter = [](const ::testing::TestParamInfo<SegmentEncodingSpec> info) {
+  const auto spec = info.param;
+
+  auto stream = std::stringstream{};
+  stream << encoding_type_to_string.left.at(spec.encoding_type);
+  if (spec.vector_compression_type) {
+    stream << "-" << vector_compression_type_to_string.left.at(*spec.vector_compression_type);
+  }
+
+  auto string = stream.str();
```

Comment:
I would suggest using another, probably more descriptive name, but definitely one which is not called like a type.